src/Microsoft.ML.Core/Data/ColumnTypeExtensions.cs

```diff
@@ -19,9 +19,19 @@ internal static class ColumnTypeExtensions
             (columnType is TimeSpanType) || (columnType is DateTimeType) || (columnType is DateTimeOffsetType);
 
         /// <summary>
-        /// Zero return means either it's not a key type or the cardinality is unknown.
+        /// Zero return means either it's not a key type.
         /// </summary>
-        public static int GetKeyCount(this ColumnType columnType) => (columnType as KeyType)?.Count ?? 0;
+        public static ulong GetKeyCount(this ColumnType columnType) => (columnType as KeyType)?.Count ?? 0;
+
+        /// <summary>
+        /// Sometimes it is necessary to cast the Count to an int. This performs overflow check.
+        /// </summary>
+        public static int AssertRangeReturnCount(this ColumnType columnType, IExceptionContext ectx = null)
+        {
+            ulong count = columnType.GetKeyCount();
+            ectx.Assert(count <= int.MaxValue, nameof(KeyType) + "." + nameof(KeyType.Count) + " exceeds int.MaxValue.");
```

Comment:
>Assert [](start = 17, length = 6)

Why is this an assert? Isn't it intended so that callers of this method don't need to do a Check?

Reply:
Yes, this *looks* like a check to me, based on how it is used. FYI if we meant an assert then there is no need for an error message, since asserts are just for code documentation.

---
In reply to: [249203459](https://github.com/dotnet/machinelearning/pull/2146#discussion_r249203459) [](ancestors = 249203459)